presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSource.java

```diff
@@ -194,41 +206,82 @@ public synchronized boolean appendToNext(PageBuilder pageBuilder, int outputChan
         }
     }
 
-    @ThreadSafe
+    /**
+     * Each thread has it's own copy of OuterPositionTracker instance and after using it,
```

Comment:
It should rather say that each LookupSource has its own OPT. And that last thread which uses it must commit it.